javaslang/src/test/java/javaslang/collection/TreeSetTest.java

```diff
@@ -39,40 +38,29 @@ protected boolean emptyShouldBeSingleton() {
 
     @Override
     protected <T> TreeSet<T> of(T element) {
-        return TreeSet.of(toStringComparator(), element);
+        return TreeSet.of(naturalComparator(), element);
+    }
+
+    @Override
+    protected <T> TreeSet<T> of(Comparator<? super T> comparator, T element) {
+        return TreeSet.of(comparator, element);
     }
 
     @SuppressWarnings("varargs")
     @SafeVarargs
     @Override
     protected final <T> TreeSet<T> of(T... elements) {
-        boolean allLongs = true;
-        boolean allNumbers = true;
-        for (T element : elements) {
-            if (!(element instanceof Number)) {
-                allNumbers = false;
-            }
-            if (!(element instanceof Long)) {
-                allLongs = false;
-            }
-        }
-        if (allLongs) {
-            return TreeSet.ofAll(toLongComparator(), Iterator.of(elements));
-        } else if (allNumbers) {
-            return TreeSet.ofAll(toDoubleComparator(), Iterator.of(elements));
-        } else {
-            return TreeSet.ofAll(toStringComparator(), Iterator.of(elements));
-        }
+        return TreeSet.<T> of(naturalComparator(), elements);
```

Comment:
Yes, under project' settings, there is a "I have my own config" checkbox which allows you to select a file from repo to use as checkstyle/findbugs/pmd config: 

![image](https://cloud.githubusercontent.com/assets/513361/20484974/eeba464c-aff9-11e6-8cf8-cd81f52686fd.png)

![image](https://cloud.githubusercontent.com/assets/513361/20485048/34c7ae90-affa-11e6-8c54-4b152dd68707.png)